tsdb/index/tsi1/partition.go

```diff
@@ -368,12 +378,15 @@ func (p *Partition) Close() error {
 	p.Wait()
 
 	// Lock index and close remaining
-	p.Mu.Lock()
-	defer p.Mu.Unlock()
+	p.mu.Lock()
+	defer p.mu.Unlock()
 
-	var err error
+	if p.fileSet == nil {
+		return nil
+	}
 
 	// Close log files.
+	var err error
 	for _, f := range p.fileSet.files {
 		if localErr := f.Close(); localErr != nil {
 			err = localErr
```

Comment:
Use a `[]error` and `errors.Join()` here to return all the errors.

https://pkg.go.dev/errors#Join

Reply:
Oh thats a really cool error handling feature. Will do 👍